dhcpv4/nclient4/client.go

```diff
@@ -386,15 +389,15 @@ func (c *Client) retryFn(fn func(timeout time.Duration) error) error {
 	timeout := c.timeout
 
 	// Each retry takes the amount of timeout at worst.
-	for i := 0; i < c.retry || c.retry < 0; i++ {
+	for i := uint(0); int(i) < c.retries || c.retries <= 0; i++ {
 		switch err := fn(timeout); err {
 		case nil:
 			// Got it!
 			return nil
 
 		case errDeadlineExceeded:
-			// Double timeout, then retry.
-			timeout *= 2
+			// Exponential backoff.
+			timeout = time.Duration(rand.Intn(1<<i)-1) * c.timeout
```

Comment:
Isn't the timeout always negative for the first iteration (when `i = 0`? Is this on purpose?)